fred2/orienteditor.cpp

```diff
@@ -237,34 +258,54 @@ bool orient_editor::query_modified()
 
 void orient_editor::OnOK()
 {
-	vec3d delta, *cur_pos;
+	vec3d delta = vmd_zero_vector;
 	matrix m = vmd_identity_matrix;
-	angles ang;
 	bool change_pos = false, change_orient = false;
 
 	UpdateData(TRUE);
 
+	// save selection for future uses of this dialog
+	Select_set_relative = set_relative;
+
+	// position...
+	const vec3d &reference_pos = set_relative ? vmd_zero_vector : Objects[cur_object_index].pos;
 	// there's enough difference in our position that we're changing it
-	cur_pos = &Objects[cur_object_index].pos;
-	if (!is_close(cur_pos->xyz.x, m_position_x) || !is_close(cur_pos->xyz.y, m_position_y) || !is_close(cur_pos->xyz.z, m_position_z))
+	if (!is_close(reference_pos.xyz.x, m_position_x) || !is_close(reference_pos.xyz.y, m_position_y) || !is_close(reference_pos.xyz.z, m_position_z))
 	{
 		vec3d pos;
 		pos.xyz.x = convert(m_position_x);
 		pos.xyz.y = convert(m_position_y);
 		pos.xyz.z = convert(m_position_z);
-		vm_vec_sub(&delta, &pos, cur_pos);
+
+		if (set_relative)
+			vm_vec_unrotate(&delta, &pos, &Objects[cur_object_index].orient);
```

Comment:
Should this not involve a subtraction too? If someone sets a ship to be relative 500 X, should it move further away in the relative X direction by 500 each time, or just set its position to be equal to 500 X relative?

Reply:
It should move 500 units in the X direction in the ship's frame of reference.  That would be a delta to whatever the ship's current position.  For example, adding 25 units in the ship's Z direction repeatedly will make the ship move forward by increments of 25.

Reply:
I see, I misunderstood this would be relative to a different ship.